### ISSUE TITLE
train_batch_size and batch_size when using multiple GPUs

### ISSUE THREAD

**Issue by ajd12342** [s3prl/s3prl#247]
Hi, I am trying to make use of multiple GPUs for running the `asr` downstream task. My commands are
`distributed="-m torch.distributed.launch --nproc_per_node 2";`
`python3 $distributed run_downstream.py -m train -u hubert_local -k /data1/scratch/anujd/complete-expts/fb-ckpts/hubert_base_ls960.pt -d asr -n hubert_asr -o "config.optimizer.lr=1.0e-5"`

Since I am using 2 GPUs, I changed the `config.runner.total_steps` to 100000 from 200000 in the asr `config.yaml`.

Now, my question is that in the asr `config.yaml`, the `train_batch_size` and `batch_size` is specified as 32. When I use 2 GPUs, will this be per-GPU or added across GPUs? Will the effective batch_size seen at each step be 32 or 64?

**Comment by leo19941227** [comment]
Hi! Thanks for the great question!
The config file for `run_downstream.py` can be considered as `configuring a single process` in the DDP training, including the batch size and the learning rate.
Hence, the effective batch size of your command will be 64.

**Comment by ajd12342** [comment]
Thank you for the quick and informative reply! I can close this issue now.
